sdk/appconfiguration/Azure.Data.AppConfiguration/src/SettingSelector.cs

```diff
@@ -17,58 +19,223 @@ namespace Azure.Data.AppConfiguration
+            AppendSeparator(Keys);
+            AppendAny(Keys);
+            AppendEscaped(Keys, key);
+            AppendAny(Keys);
+            return this;
+        }
 
         /// <summary>
-        /// Creates a <see cref="SettingSelector"/> that will retrieve <see cref="ConfigurationSetting"/> entities that match
-        /// the passed-in keys and labels.
         /// </summary>
-        /// <param name="key">A key or key filter indicating which <see cref="ConfigurationSetting"/> entities to select.</param>
-        /// <param name="label">A label or label filter indicating which <see cref="ConfigurationSetting"/> entities to select</param>
-        public SettingSelector(string key, string label = default)
+        /// <param name="label"></param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        public SettingSelector AddLabel(string label)
+        {
+            Argument.AssertNotNull(label, nameof(label));
+            AppendSeparator(Labels);
+            AppendEscaped(Labels, label.Length > 0 ? label : "\0");
+            return this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="label"></param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        public SettingSelector AddLabelStartsWith(string label)
+        {
+            Argument.AssertNotNullOrEmpty(label, nameof(label));
+            AppendSeparator(Labels);
+            AppendEscaped(Labels, label);
+            AppendAny(Labels);
+            return this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="label"></param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        public SettingSelector AddLabelEndsWith(string label)
+        {
+            Argument.AssertNotNullOrEmpty(label, nameof(label));
+            AppendSeparator(Labels);
+            AppendAny(Labels);
+            AppendEscaped(Labels, label);
+            return this;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="label"></param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        public SettingSelector AddLabelContains(string label)
+        {
+            Argument.AssertNotNullOrEmpty(label, nameof(label));
+            AppendSeparator(Labels);
+            AppendAny(Labels);
+            AppendEscaped(Labels, label);
+            AppendAny(Labels);
+            return this;
+        }
+
+        /// <summary>
+        /// The fields of the <see cref="ConfigurationSetting"/> to retrieve for each setting in the retrieved group.
+        /// </summary>
+        /// <param name="fields">Fields to be returned.</param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        public SettingSelector SetFields(SettingFields fields)
```

Comment:
Why is this a Set method rather than a settable property?

Reply:
I didn't think user needs to read the field. But maybe it can be needed for a scenario where flag value is added to existing one.